src/metaswitch/clearwater/config_manager/config_access.py

```diff
@@ -343,19 +353,48 @@ def main(args):
         try:
             upload_verified_config(config_loader,
                                    local_store,
-                                   args.config_type,
+                                   selected_config,
                                    args.force,
                                    args.autoconfirm)
         except (UserAbort, ConfigUploadFailed) as exc:
             log.error("Upload failed")
             sys.exit(exc)
 
 
+def lookup_config_type(config_type, config_location):
+    """
+    This takes the string defining the config_type from arg.parse and returns
+    a instance of the class containing data on the config type selected
+    """
+    config_classes = load_plugins_in_dir(PLUGIN_DIR, config_location)
+    for choice in config_classes:
+        if config_type == choice.name:
+            return choice
+        # The config_type will always be the choice.name of one class as
+        # parse_arguments will make sure config_type == choice.name
+
+
 def parse_arguments():
     """
     Parse the arguments passed to the script.
+    The options for config type are designated by the plugins available
     :return:
     """
+    # Pick the options to appear in argument parsing, the config types are
+    # loaded as plugin classes onto the node so only the config files available
+    # on the node are able to be changed on the node.
+    config_classes = load_plugins_in_dir(PLUGIN_DIR)
+
+    # If there are no config_classes loaded from the PLUGIN_DIR this means you
+    # will be unable to select anything, hence you want to raise an error, that
+    # makes sense. Otherwise you would get always get refused by the arg
+    # parsing.
```

Comment:
Good pattern to try to catch yourself doing and fix: whenever you're writing a comment above an if statement and the comment includes a restatement of that if statement, it's almost always better to remove the restatement and move the comment inside the if block. Eg. rather than:
```
# If x is non-empty, do something
if x:
    do_something()
```
Instead write:
```
if x:
    # We have to do something because ...
    do_something()
```